### DISCUSSION TITLE
How to synchronize tiptap content to my server?

### DISCUSSION THREAD

**Discussion by Luciennnnnnn** [ueberdosis/tiptap#3010] [General Questions]
Is Y.js a suitable solution for that? I have no multi-user collaborative requirement.

I know that I can listen to `onUpdate` event and use `editor.getJSON()` to get changed content, then I can set this content through an API to the server. Then the server updates a field of MongoDB with the new value(I may store it as a JSON string).

I have a few concerns about the above simple design choice:

If the user only updates a small piece of document, I have to send full document JSON to the server and update the full content of the MongoDB field; all of these seem too costly.

In the frontend, I may use a throttle to reduce trigger frequency, but I still have to send full JSON. It may be a bottleneck as content increases with more text and images.

Can I only send the content change to the server like collaborative applications? But I have no requirement for collaborative editing between users. I only want to support fast client/server synchronization. Is Y.js a proper solution for me? If there are some better tools for me, please let me know.

And if we could send the change of content, can we update the partial JSON string on MongoDB? The content of tiptap may be a complex nested JSON. Is it possible?

**Comment by aehlke** [comment]
https://hocuspocus.dev
